inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -501,12 +501,28 @@ void GNAGraphCompiler::PowerPrimitive(InferenceEngine::CNNLayerPtr layer) {
     auto input = layer->insData[0].lock();
 
     auto outputs = *layer->outData.begin();
-
     uint32_t num_rows_in = FROM_IR_DIM(input, 1);
     uint32_t num_columns_in = FROM_IR_DIM(input, 2);
     uint32_t num_rows_out = num_rows_in;
     uint32_t num_padding = ALIGN(num_rows_in, 8) - num_rows_in;
 
+    if (num_columns_in > 8) {
+        auto PrevFunctionalLayer = [](CNNLayerPtr l, int idx = 0) {
+            auto prevLayer = CNNNetPrevLayerSkipCertain(l, idx, [](CNNLayerPtr ptr) {
+                return LayerInfo(ptr).isNonFunctional();
+            });
+            gnalog() << "CNNNetPrevLayerSkipCertain for :: " << l->name << "returned: " << prevLayer->name << std::endl;
+            return prevLayer;
+        };
+        auto prevLayer = PrevFunctionalLayer(layer, 0);
+        auto new_output = *prevLayer->outData.begin();
+        num_rows_in = FROM_IR_DIM(new_output, 1);
+        num_columns_in = FROM_IR_DIM(new_output, 2);
+        input->reshape({num_columns_in, num_rows_in}, Layout::CN);
```

Comment:
please explain how this solve the problem, what if previous functional layer has 4d output for example?